my-app/src/components/championsGrid/ChampionsGrid.tsx

```diff
@@ -15,17 +14,59 @@ import {
 	sortAscending,
 	sortDescending,
 } from "../../features/champions/getChampions";
+import Champion from "../../features/champions/types/champion";
+import ToastContainerComponent from "../../components/toasts/toastContainer";
 
 const PaginatedItems = (props: any) => {
 	const dispatch = useDispatch();
 	const navigate = useNavigate();
 	const { id } = useParams();
 	const { page } = useParams();
 	const [show, setShow] = React.useState(false);
-	const [modalChampion, setModalChampion] = React.useState(0);
+	const [champion, setChampion] = React.useState<any>();
 	const reduxState = useSelector((state: any) => state.champions);
+	const watchlistState = useSelector((state: any) => state.watchlist);
 	const { loading, error, champions, totalPages } = reduxState;
 
+	const handleCloseModal = () => {
+		setShow(false);
+		navigate(`/${page}`);
+	};
+
+	const handlePageClick = (data: any) => {
+		fetchApi(dispatch, data.selected + 1, props.itemsPerPage);
+		navigate(`/${data.selected + 1}`);
+	};
+
+	const modalOpen = (champion: Champion, championID: number) => {
+		setChampion(champion);
+		setShow(true);
+		navigate(`/${page}/${championID}`);
+	};
+
+	const checkUrl = () => {
+		if (id) {
+			setShow(true);
+			setChampion(
+				champions.find((champion: any) => champion.id === Number(id))
+			);
+		}
+	};
+
+	const isSaved = (championID: number) => {
+		return watchlistState.champions.findIndex(
+			(champion: any) => champion.id === championID
+		);
+	};
+
+	useEffect(() => {
+		fetchApi(dispatch, Number(page), props.itemsPerPage);
+	}, [id || page]);
```

Comment:
How is this different from:

```suggestion
	}, [id, page]);
```

Reply:
I guess its not, but when things didnt work because the useEffect wasn't triggering i had to try them all xD